util/src/main/java/com/google/cloud/hadoop/util/interceptors/InvocationIdInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package com.google.cloud.hadoop.util.interceptors;
+
+import com.google.api.client.http.HttpExecuteInterceptor;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import java.io.IOException;
+import java.util.UUID;
+
+public final class InvocationIdInterceptor implements HttpExecuteInterceptor {
```

Comment:
can this be not private.

Reply:
Also add javadoc

Reply:
added java doc.
It's a `public` class.

Reply:
I meant, can it be not public

Reply:
No.

Reply:
Please add @VisibleForTesting annotation to indicate that the class is public only for testing purposes.

Reply:
Sure.